### PR TITLE
DPE-5210 Common credentials fixture and `exec` timeout workaround

### PR DESCRIPTION
**Pull request by paulomach**
## Issue

* Freeze DB test fails when waiting for credentials.
* multiple calls for same credentials during testing
* ops's `container.exec` timeout is not consistently working

## Solution

* Fixture for credentials retrieval
* workaround for `container.exec` timeout

## TODO

~~Propagate lib change back to VM~~

### CODE REVIEW

File: lib/charms/mysql/v0/mysql.py [hunk 1]
```diff
@@ -1536,7 +1536,9 @@ def cluster_metadata_exists(self, from_instance: str) -> bool:
         )
 
         try:
-            output = self._run_mysqlsh_script("\n".join(check_cluster_metadata_commands))
+            output = self._run_mysqlsh_script(
+                "\n".join(check_cluster_metadata_commands), timeout=10
+            )
```
Comment (paulomach) [inline lib/charms/mysql/v0/mysql.py:1541]: _Note to self_
* libpatch bump
* merge on vm

File: tests/integration/high_availability/test_self_healing.py [hunk 2]
```diff
@@ -97,38 +97,41 @@ async def test_kill_db_process(
...
+    await ensure_all_units_continuous_writes_incrementing(ops_test, credentials=credentials)
 
     mysql_unit = ops_test.model.applications[mysql_application_name].units[0]
     primary = await get_primary_unit(ops_test, mysql_unit, mysql_application_name)
 
     mysql_pid = await get_process_pid(
         ops_test, primary.name, MYSQL_CONTAINER_NAME, MYSQLD_PROCESS_NAME
     )
-    assert mysql_pid > 0, "mysql process id is not positive"
+    assert (mysql_pid or -1) > 0, "mysql process id is not positive"
```
Comment (taurus-forever) [inline tests/integration/high_availability/test_self_healing.py:134]: 1) When can PID be missing?
2) below the error message is `is not known/positive` but here is just `not positive`. On purpose?

**Review by taurus-forever — APPROVED** [review]
Yeah, timeout 10 seconds should be enough for all case (c) Bill Gates.
Nice, I suspect caching credentials will speedup the all tests and finally removes pink from allure. Nice catch with container.exec() !

File: tests/integration/high_availability/test_self_healing.py (hunk 2 above)
Comment (paulomach) [reply tests/integration/high_availability/test_self_healing.py:134]: this is just to quiet down the linter. the pid will be `None` when not found, causing linter messages on further use of the `mysql_pid` variable, which is very annoying

**Review by shayancanonical — APPROVED** [review]
awesome catches!

**Review by taurus-forever — APPROVED** [review]
Excellent! Thank you!

**Pull request merged**